### PR TITLE
10324 missing ability specs

### PR DESCRIPTION
**Pull request by smoyth**
- Added ability specs for QingGroup, OptionSet, OptionNode
- Consolidated OptionSet `change_geographic` under `update_core`
- Refactored some older code
- Saved this [gist of ability usages](https://gist.github.com/cooperka/47f50bd17f854e25ec179b382d9e9e4f) for future refactoring (ideally it will remain relevant as long as we maintain `ability.rb` during future changes, but I also included the regex for finding all usages in case we want to run again)

TODO (Tom):
- More cleanup

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -49,12 +49,15 @@ def to_s
...
   # and can prevent the definition of a scope for use with accessible_by.
   # So we are moving away from using :manage, preferring to grant permissions explicitly instead.
+  #
   # When refactoring away from :manage, be sure to account for all permissions that may be getting checked
   # at runtime. Places to look are:
   # - Throughout this file
-  # - Throughout the app by grepping can?, cannot?, and accessible_by
-  # - The routes file (are there any custom actions combined authorize_resource?-if so, consider using
-  #   skip_authorize_resource and using a stock permission unless the custom one is really needed.)
+  # - Throughout this gist https://gist.github.com/cooperka/47f50bd17f854e25ec179b382d9e9e4f
```
Comment (smoyth) [inline app/models/ability.rb]: ❤️

File: spec/models/ability/form_ability_spec.rb [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
   let(:object) { form }
   let(:all) do
-    %i[add_questions change_status clone destroy download remove_questions reorder_questions show update]
+    Ability::CRUD + %i[add_questions change_status clone download remove_questions reorder_questions]
```
Comment (smoyth) [inline spec/models/ability/form_ability_spec.rb]: Cool

File: spec/models/ability/qing_group_ability_spec.rb [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        let(:form) { create(:form, :live, question_types: %w[text]) }
+        let(:permitted) { all }
+        it_behaves_like "has specified abilities"
+      end
+
+      context "when standard" do
+        let(:form) { create(:form, :standard, question_types: %w[text]) }
+        let(:permitted) { [] }
+        it_behaves_like "has specified abilities"
+      end
```
Comment (smoyth) [inline spec/models/ability/qing_group_ability_spec.rb]: Not sure if it makes sense to test standard stuff when in mission mode? Is this done elsewhere?

Comment (cooperka) [reply spec/models/ability/qing_group_ability_spec.rb]: I think the intent here is to test a **coordinator** trying to gain access to a standard object (admin standard mode is tested separately with an admin user). This was copied from a pre-existing spec.

File: app/models/option_node.rb [hunk 4]
```diff
@@ -242,6 +242,8 @@ def arrange_as_json(hash = nil)
         # And option_has_answers? kicks off a big SQL query for a huge set.
         # Conditions association should be eager loaded.
         unless huge?
+          # This should ideally consult the `destroy` permission for option_node in the authorization system,
+          # but it all needs to be refactored anyway; it's not worth doing more gross things just for this.
           branch[:removable] = !option_set.option_has_answers?(node.option_id) && node.conditions.empty?
```
Comment (cooperka) [inline app/models/option_node.rb]: Is this comment still relevant? You asked me to document this while pairing but I'm not completely sure what it's referring to.

**Comment by cooperka** [comment]
@smoyth I can't assign you to review since you submitted this PR, but it's now ready for your review.

**Comment by cooperka** [comment]
Closing in favor of #721

**Pull request closed without merging**